core/templates/pages/diagnostic-test-player-page/diagnostic-test-player.component.ts

```diff
@@ -78,10 +104,26 @@ export class DiagnosticTestPlayerComponent implements OnInit {
     );
 
     this.getProgressText();
+
     this.classroomBackendApiService
-      .getClassroomIdAsync(this.classroomUrlFragment)
-      .then(classroomId => {
-        this.classroomId = classroomId;
+      .fetchClassroomDataAsync(this.classroomUrlFragment)
+      .then(classroomData => {
+        this.classroomData = classroomData;
+      })
+      .catch(error => {
+        this.isStartTestButtonDisabled = true;
+        if (error.status === 500) {
```

Comment:
This inconsistency is strange. For 500 we route to an error page, for others we show a warning. IIRC warnings are the default approach in Oppia to tell about the problem, so let's stick to it

Reply:
Done. let's just show warning and disable start button.